Tests/GRPCProtobufCodeGenTests/ProtobufCodeGeneratorTests.swift

```diff
@@ -0,0 +1,425 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if os(macOS) || os(Linux)  // swift-format doesn't like canImport(Foundation.Process)
+
+import GRPCCodeGen
+import GRPCProtobufCodeGen
+import XCTest
+
+final class ProtobufCodeGeneratorTests: XCTestCase {
+  func testProtobufCodeGenerator() throws {
+    try testCodeGeneration(
+      indentation: 4,
+      visibility: .internal,
+      client: true,
+      server: false,
+      expectedCode: """
+        // Copyright 2015 gRPC authors.
+        //
+        // Licensed under the Apache License, Version 2.0 (the "License");
+        // you may not use this file except in compliance with the License.
+        // You may obtain a copy of the License at
+        //
+        //     http://www.apache.org/licenses/LICENSE-2.0
+        //
+        // Unless required by applicable law or agreed to in writing, software
+        // distributed under the License is distributed on an "AS IS" BASIS,
+        // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        // See the License for the specific language governing permissions and
+        // limitations under the License.
+
+        // DO NOT EDIT.
+        // swift-format-ignore-file
+        //
+        // Generated by the gRPC Swift generator plugin for the protocol buffer compiler.
+        // Source: helloworld.proto
+        //
+        // For information on using the generated types, please see the documentation:
+        //   https://github.com/grpc/grpc-swift
+
+        import GRPCCore
+        import GRPCProtobuf
+        internal enum Helloworld {
+            internal enum Greeter {
+                internal enum Methods {
+                    internal enum SayHello {
+                        internal typealias Input = HelloRequest
+                        internal typealias Output = HelloReply
+                        internal static let descriptor = MethodDescriptor(
+                            service: "helloworld.Greeter",
+                            method: "SayHello"
+                        )
+                    }
+                }
+                internal static let methods: [MethodDescriptor] = [
+                    Methods.SayHello.descriptor
+                ]
+                internal typealias ClientProtocol = Helloworld_GreeterClientProtocol
+                internal typealias Client = Helloworld_GreeterClient
+            }
+        }
+
+        /// The greeting service definition.
+
+        internal protocol Helloworld_GreeterClientProtocol: Sendable {
```

Comment:
There shouldn't be a blank line between these.

Reply:
Yes, it's in the to do list from the description - I will modify the renderer in a different PR to avoid this blank line

Reply:
Oops, I sorry, missed that.